### PR TITLE
adds unit tests for Overview-related objects

### PR DESCRIPTION
**Pull request by bamatson**

### CODE REVIEW

File: app/src/main/java/io/mattcarroll/androidtesting/accounts/RandomTransactions.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
             "Chick fil A",
             "Venmo",
             "Trejo's Tacos",
+            "Trejo's Donuts",
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/accounts/RandomTransactions.java]: Thank God we didn't forget this one!

File: app/src/main/java/io/mattcarroll/androidtesting/overview/Time.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        Calendar yesterday = Calendar.getInstance();
+        yesterday.add(Calendar.DAY_OF_YEAR, -1);
+
+        Calendar otherDay = Calendar.getInstance();
+        otherDay.setTime(new Date(date));
+
+        return isSameYear(yesterday, otherDay) && isSameDayOfYear(yesterday, otherDay);
+    }
+
+    public boolean isOnSameDay(long date1, long date2) {
```
Comment (matthew-carroll) [inline app/src/main/java/io/mattcarroll/androidtesting/overview/Time.java]: Can't you implement itToday() and isYesterday() but passing today's time and yesterday's time intot his method?

File: app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import io.mattcarroll.androidtesting.accounts.BankAccount;
+import io.mattcarroll.androidtesting.accounts.Transaction;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (matthew-carroll) [inline app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java]: Doesn't look like you need this runner.

File: app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        String expectedDisplayName = "Acme Super Deluxe Card";
+
+        List<AccountViewModel> viewModels = presenter.present(accounts);
+
+        String displayName = viewModels.get(0).getDisplayName();
+        assertEquals(expectedDisplayName, displayName);
+    }
+
+    @Test
+    public void itPresentsCorrectBalanceGivenTransactions() {
```
Comment (matthew-carroll) [inline app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java]: A better name might be 'itPresentsBalanceCalculatedFromTransactions'.  I think being "Correct" is a given.

File: app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        String expectedBalance = "$0.00";
+
+        List<AccountViewModel> viewModels = presenter.present(accounts);
+
+        String balance = viewModels.get(0).getBalance();
+        assertEquals(expectedBalance, balance);
+    }
+
+    @Test
+    public void itPresentsCorrectAmountSpentGivenTransactions() {
```
Comment (matthew-carroll) [inline app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java]: Same here with "Correct"

File: app/src/test/java/io/mattcarroll/androidtesting/overview/TransactionListPresenterTest.java [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    private static long daysAgo(int days) {
+        Calendar calendar = new GregorianCalendar();
+        calendar.setTime(new Date(TODAY));
+        calendar.add(Calendar.DATE, -1 * days);
+        return calendar.getTimeInMillis();
+    }
+
+    private TransactionListPresenter presenter;
+
+    @Mock
```
Comment (matthew-carroll) [inline app/src/test/java/io/mattcarroll/androidtesting/overview/TransactionListPresenterTest.java]: Can you create a 2nd version without using Mockito so we can ask them to write this before we get to Mockito?

File: app/src/test/java/io/mattcarroll/androidtesting/overview/TransactionListPresenterTest.java [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        List<Transaction> transactions = transactionsWithDescriptions(DESCRIPTION);
+
+        List<Object> viewModels = presenter.present(transactions);
+
+        TransactionListItemViewModel viewModel = (TransactionListItemViewModel) viewModels.get(1);
+        assertEquals(DESCRIPTION, viewModel.title());
+    }
+
+    @Test
+    public void itPresentsTransactionAmountAsDetail() {
```
Comment (matthew-carroll) [inline app/src/test/java/io/mattcarroll/androidtesting/overview/TransactionListPresenterTest.java]: this test looks the same as the previous one?

File: app/src/main/java/io/mattcarroll/androidtesting/accounts/RandomTransactions.java (hunk 1 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/accounts/RandomTransactions.java]: just opened bro. this database needs to be fresh!

File: app/src/main/java/io/mattcarroll/androidtesting/overview/Time.java (hunk 2 above)
Comment (bamatson) [inline app/src/main/java/io/mattcarroll/androidtesting/overview/Time.java]: indeed. I'll do that

File: app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java (hunk 4 above)
Comment (bamatson) [inline app/src/test/java/io/mattcarroll/androidtesting/overview/AccountsPresenterTest.java]: fair

File: app/src/test/java/io/mattcarroll/androidtesting/overview/TransactionListPresenterTest.java (hunk 7 above)
Comment (bamatson) [inline app/src/test/java/io/mattcarroll/androidtesting/overview/TransactionListPresenterTest.java]: yep, forgot to implement that one

**Pull request merged**